vcr/api/v2/registry.go

```diff
@@ -61,110 +61,94 @@ func (w *Wrapper) SearchVCs(ctx echo.Context) error {
+				searchTerms[i] = st
+				i++
+			}
+		case float64:
+			if value != 0.0 {
+				searchTerms[i] = st
+				i++
+			}
 		}
+		// TODO bool problem, luckily there are no default bool fields in a VC
 	}
+	return searchTerms[:i]
+}
 
-	results, err := w.VCR.SearchLegacy(ctx.Request().Context(), query, allowUntrusted, nil)
-	if err != nil {
-		return err
+func flatten(document interface{}, currentPath []string) []vcr.SearchTerm {
+	switch t := document.(type) {
+	case jsonld.Document:
+		return flattenSlice(t, currentPath)
+	case []interface{}:
+		return flattenSlice(t, currentPath)
+	case map[string]interface{}:
+		return flattenMap(t, currentPath)
 	}
 
-	return ctx.JSON(http.StatusOK, results)
+	return nil
 }
 
-func (w *Wrapper) searchAuths(ctx echo.Context, allowUntrusted bool, vcQuery vc.VerifiableCredential) error {
-	// credentialSubject struct that reflect current supported query params
-	type credentialSubject struct {
-		ID           string `json:"id"`
-		PurposeOfUse string `json:"purposeOfUse"`
-		Subject      string `json:"subject"`
-		Resources    struct {
-			Path string `json:"path"`
-		} `json:"resources"`
-	}
+func flattenSlice(expanded []interface{}, currentPath []string) []vcr.SearchTerm {
+	result := make([]vcr.SearchTerm, 0)
 
-	var subjectsQuery []credentialSubject
-	if err := vcQuery.UnmarshalCredentialSubject(&subjectsQuery); err != nil {
-		return core.InvalidInputError("failed to unmarshall VC subject: %w", err)
+	for _, sub := range expanded {
+		result = append(result, flatten(sub, currentPath)...)
 	}
 
-	// convert to query
-	query, err := w.VCR.Registry().QueryFor(concept.AuthorizationConcept)
-	if err != nil {
-		return err
-	}
-	if vcQuery.Issuer.String() != "" {
-		query.AddClause(concept.Eq("issuer", vcQuery.Issuer.String()))
-	}
-	if len(subjectsQuery) > 0 {
-		subjectQuery := subjectsQuery[0]
-		if subjectQuery.ID != "" {
-			query.AddClause(concept.Eq("credentialSubject.id", subjectQuery.ID))
-		}
-		if subjectQuery.PurposeOfUse != "" {
-			query.AddClause(concept.Eq("credentialSubject.purposeOfUse", subjectQuery.PurposeOfUse))
-		}
-		if subjectQuery.Subject != "" {
-			query.AddClause(concept.Eq("credentialSubject.subject", subjectQuery.Subject))
-		}
-		if subjectQuery.Resources.Path != "" {
-			query.AddClause(concept.Eq("credentialSubject.resources.#.path", subjectQuery.Resources.Path))
-		}
-	}
+	return result
+}
 
-	results, err := w.VCR.SearchLegacy(ctx.Request().Context(), query, allowUntrusted, nil)
-	if err != nil {
-		return err
+func flattenMap(expanded map[string]interface{}, currentPath []string) []vcr.SearchTerm {
+	// JSON-LD in expanded form either has @value, @id, @list or objects
+
+	results := make([]vcr.SearchTerm, 0)
+
+	for k, v := range expanded {
+		switch k {
+		case "@id":
+			fallthrough
+		case "@value":
+			results = append(results, vcr.SearchTerm{
+				IRIPath: currentPath,
+				Value:   v,
+			})
+		case "@list":
+			// TODO: not supported... This would require OR type queries
```

Comment:
maybe return an error indicating the query isn't supported? Otherwise it might lead to the querier falsely assuming there are no VCs with the given parameters?

Reply:
actually added every item in the list as required match. This would still be according to spec (all must match)